crates/turbopack-core/src/chunk/dev.rs

```diff
@@ -107,27 +110,73 @@ impl ChunkingContext for DevChunkingContext {
     }
 
     #[turbo_tasks::function]
-    async fn chunk_path(
-        &self,
-        path_vc: FileSystemPathVc,
-        extension: &str,
-    ) -> Result<FileSystemPathVc> {
+    async fn chunk_path(&self, ident: AssetIdentVc, extension: &str) -> Result<FileSystemPathVc> {
         fn clean(s: &str) -> String {
             s.replace('/', "_")
         }
-        // For clippy -- This explicit deref is necessary
-        let path = &*path_vc.await?;
+        let ident = &*ident.await?;
 
+        // For clippy -- This explicit deref is necessary
+        let path = &*ident.path.await?;
         let mut name = if let Some(inner) = self.context_path.await?.get_path_to(path) {
             clean(inner)
         } else {
-            clean(&path_vc.to_string().await?)
+            clean(&ident.path.to_string().await?)
         };
         let removed_extension = name.ends_with(extension);
         if removed_extension {
             name.truncate(name.len() - extension.len());
         }
 
+        if !ident.params.is_empty() {
+            let default_modifier = match extension {
+                ".js" => Some("ecmascript"),
+                ".css" => Some("css"),
+                _ => None,
+            };
+
+            let mut hasher = Xxh3Hash64Hasher::new();
+            let mut has_hash = false;
+            for param in ident.params.iter() {
+                match *param {
+                    AssetParam::Query(query) => {
+                        0_u8.deterministic_hash(&mut hasher);
```

Comment:
That won't work here well since it is async. We would need a `AsyncDeterministicHash` where `deterministic_hash` is an async method. That's a lot work...